src/fe-gtk/fe-gtk.c

```diff
@@ -293,10 +293,10 @@ create_input_style (GtkStyle *style)
 		GtkSettings *settings = gtk_settings_get_default ();
 		char *theme_name;
 
-		/* gnome-themes-standard 3.20 relies on images to do theming
+		/* gnome-themes-standard 3.20+ relies on images to do theming
 		 * so we have to override that. */
 		g_object_get (settings, "gtk-theme-name", &theme_name, NULL);
-		if (g_str_has_prefix (theme_name, "Adwaita"))
```

Comment:
Can you test Yaru-dark also and change it to a prefix check like Adwaita?

Reply:
I ran `gsettings set org.gnome.desktop.interface gtk-theme Yaru-dark` and confirmed application response (Chromium turned dark, text boxes in question did too, but lost their theme). Changing to a prefix check resolves this, and maintains the theme through a gtk-theme update while Hexchat is open. I've also updated the commit message.